table.cc

```diff
@@ -962,6 +962,17 @@ void table::trigger_offstrategy_compaction() {
     _compaction_manager.submit_offstrategy(this);
 }
 
+future<> table::submit_offstrategy_compaction() {
+    // If the user calls trigger_offstrategy_compaction() to trigger
+    // off-strategy explicitly, cancel the timeout based automatic trigger.
+    if (!_off_strategy_trigger.armed()) {
+        return make_ready_future<>();
+    }
+
+    _off_strategy_trigger.cancel();
+    return _compaction_manager.do_submit_offstrategy(this);
```

Comment:
I missed this: do_submit_offstrategy(). functions with this name is usually confirmed to private namespace of a class. given that this is about running / performing offstrategy, I'd like it to have a more descriptive name. Following the name convention of compaction_manager, I  think perform_offstrategy() would be a good one.

Reply:
Fixed in v4.